### PR TITLE
Introduce a translation:check-missing command

### LINKED ISSUE #388: Add a command to check if all extractable translations exist in the translation files
For CI purposes it would be nice to have an easy way to check if there are any translation keys found in the source code that are missing in the translation file(s) for a particular locale. The command should return a non-zero exit code if translations are missing. Ideally it would also fail on translations that exist but only contain an empty string.

For example:

If one of my Twig templates contains the following:

```twig
{{ 'message.foo'|trans }}
```

But `translations/messages.en.xlf` does NOT contain `message.foo`, and I run `bin/console translation:check en`, it should output that the translation is missing and return a non-zero exit code.

I could then run this check for instance in pull request builds to enforce that the author of the pull request provides an English source translation.

Our current solution to achieve this is based on the existing `bin/console translation:extract`:

```bash
exit $(expr $(bin/console translation:extract --hide-errors | grep 'New messages' | cut -c25-28))
```

but this is obviously quite fragile and will break if the output of `translation:extract` is changed. Also, if the translation of `message.foo` already exists but is an empty string, it doesn't break the build.

### PR DESCRIPTION
**Pull request by nicwortel**
Fixes #388.

### CODE REVIEW

**Comment by nicwortel** [comment]
I don't believe the PHPStan error is introduced by my pull request.

>  ```
> -- ------------------------------------------------------------------------- 
>      Error                                                                    
>  -- ------------------------------------------------------------------------- 
>      Ignored error pattern #^Call to method getValue\(\) on an unknown class  
>      Symfony\\Component\\Translation\\DataCollector\\Data\.$# in path         
>      /github/workspace/Controller/SymfonyProfilerController.php was not       
>      matched in reported errors.                                              
>  -- ------------------------------------------------------------------------- 
> ```

I believe this is caused by an ignored error (in `phpstan-baseline.neon`) that is no longer reported. I could remove the offending lines from `phpstan-baseline.neon`.

**Comment by Guite** [comment]
> I could remove the offending lines from phpstan-baseline.neon.

Looks like it's removed as part of #396 here: https://github.com/php-translation/symfony-bundle/pull/396/files#diff-cbcd91b5fa0a393ebd972f66cc47dc90

**Comment by nicwortel** [comment]
@Guite you are right. #396 is closed but #395 has fixed the build, so after merging master all checks are green.

My PR is now ready to be reviewed.

File: Command/CheckCommand.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+use Symfony\Component\Translation\MessageCatalogueInterface;
+use Translation\Bundle\Catalogue\CatalogueCounter;
+use Translation\Bundle\Catalogue\CatalogueFetcher;
+use Translation\Bundle\Model\Configuration;
+use Translation\Bundle\Service\ConfigurationManager;
+use Translation\Bundle\Service\Importer;
+
+final class CheckCommand extends Command
+{
+    protected static $defaultName = 'translation:check';
```
Comment (welcoMattic) [inline Command/CheckCommand.php]: IMHO this name is too generic. What are we checking? What about `translation:check:extracted` or something more precise?

Comment (nicwortel) [reply Command/CheckCommand.php]: Sure, that's fine by me! However, all other commands exist of two words (`translation:download`, `translation:extract`) or are separated by a dash (`translation:delete-obsolete`). Doesn't this make it inconsistent?

Also the intention of this command is to check for missing translations (either missing as in not extracted or missing as in empty) so an alternative option would be `translation:check-missing`.

What do you prefer?

Comment (rvanlaak) [reply Command/CheckCommand.php]: Imo `translation:check-missing` and renaming the class to `CheckMissingCommand` is sufficient 👍

Comment (welcoMattic) [reply Command/CheckCommand.php]: `translation:check-missing` LGTM! Thanks

**Comment by welcoMattic** [comment]
@php-translation/deciders ok for you? We could target a 0.12 with this new command?

File: Command/CheckMissingCommand.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        $this->configurationManager = $configurationManager;
+        $this->catalogueFetcher = $catalogueFetcher;
+        $this->importer = $importer;
+        $this->catalogueCounter = $catalogueCounter;
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setName(self::$defaultName)
```
Comment (odolbeau) [inline Command/CheckMissingCommand.php]: I think it's not needed when defining the static property.

Comment (nicwortel) [reply Command/CheckMissingCommand.php]: I believe you are right, the [Symfony documentation example also doesn't call `setName()` anymore](https://symfony.com/doc/current/console.html#configuring-the-command).

However, I copied this from the other commands in this bundle which also call `->setName(self::$defaultName)`. So perhaps this should be fixed for all commands at once in a separate PR.

**Comment by nicwortel** [comment]
Are we waiting for more approvals, or is there anything I should do to get this merged? :)

**Comment by welcoMattic** [comment]
@nicwortel it's merged, we will release the 0.12 as soon as possible :wink:

**Comment by nicwortel** [comment]
Nice :+1: I will try to find some time to create a PR for https://github.com/php-translation/documentation.

**Comment by welcoMattic** [comment]
Thanks!

**Pull request merged**